### PR TITLE
[ThumbTrack] Discrete dots scale slowly with track height.

### LINKED ISSUE #8741: [Slider] Tick marks overlap when track height is increased
Increasing `trackHeight` on a discrete `MDCSlider` causes the thumb track's tick marks to overlap. Design will probably need to provide guidance regarding how the tick marks should be rendered when expanded. 

## Reproduction steps

1. Create a discrete slider
2. Increase `trackHeight` to 6
3. Drag the thumb

## Expected behavior

Tick marks should be distinct and non-overlapping.

## Actual behavior

Tick marks overlap (see screenshot)

testDiscreteSliderLargerTrackHeight_11_2@2x

### PR DESCRIPTION
**Pull request by romoore**
The discrete dots in the ThumbTrack originally assumed that the track would
only ever be 2 points high. Now that customizable heights are allowed, the
dots should scale more slowly.

This is still not perfect, since the original positioning of the ticks resulted in the filled track stopping at non-center positions of each tick. However, it is an incremental improvement and better VX will necessitate Material Design inputs.

Example showing misaligned track and tick mark.

Improves #8741

### CODE REVIEW

File: components/Slider/tests/snapshot/MDCSliderSnapshotTests.m [hunk 1]
```diff
@@ -55,7 +55,7 @@ - (void)setUp {
 
   // Uncomment below to recreate all the goldens (or add the following line to the specific
   // test you wish to recreate the golden for).
-  // self.recordMode = YES;
+  //  self.recordMode = YES;
```
Comment (bryanoltman) [inline components/Slider/tests/snapshot/MDCSliderSnapshotTests.m]: Nit: revert this

File: components/private/ThumbTrack/src/MDCThumbTrack.m [hunk 2]
```diff
@@ -104,10 +104,24 @@ - (void)drawRect:(CGRect)rect {
...
     CGContextRef contextRef = UIGraphicsGetCurrentContext();
 
-    CGRect circleRect = CGRectMake(0, 0, self.bounds.size.height, self.bounds.size.height);
+    // The dot should be:
+    // +  (1, 1) at trackHeight == 1
+    // +  (2, 2) at trackHeight == 2
+    // +  (2, 2) at trackHeight == 4
+    // +  (2.5, 2.5) at trackHeight == 5
+    // +  (3, 3) at trackHeight == 6
+    // +  ... and so on.
```
Comment (bryanoltman) [inline components/private/ThumbTrack/src/MDCThumbTrack.m]: These examples are very useful, but adding a more general explanation of what is happening here would be also helpful so the reader doesn't have to derive the logic from the examples (this needs to be wordsmithed a bit, but something like "for trackHeight < 4, the tick marks are dots with a 2pt diameter. For track heights > 4, the tick marks become rounded rects with height of trackHeight/2"). Also, the size values for trackHeight > 4 are not correct – at trackHeight 6, the dot is not (3,3), but (2,3).

**Review by bryanoltman — CHANGES_REQUESTED** [review]
Code looks great, but some of the (excellent) documentation contains incorrect info

File: components/Slider/tests/snapshot/MDCSliderSnapshotTests.m (hunk 1 above)
Comment (romoore) [reply components/Slider/tests/snapshot/MDCSliderSnapshotTests.m]: You're the second person who didn't like moving to the standard formatting for these comments.  Could you help me understand why you want this particular formatting (one space)?

File: components/private/ThumbTrack/src/MDCThumbTrack.m (hunk 2 above)
Comment (romoore) [reply components/private/ThumbTrack/src/MDCThumbTrack.m]: Done, please take a look at the latest version.

**Review by bryanoltman — APPROVED** [review]

**Pull request merged**